spec/support/job_shared_examples.rb

```diff
@@ -7,31 +7,41 @@
 end
 
 shared_examples 'an ElasticsearchIndexJob' do |container_sym|
+  it {
+    expect{described_class.perform_now}.to raise_error(ArgumentError)
+  }
   before {
     ActiveJob::Base.queue_adapter = :test
   }
   include_context 'with job runner', described_class
 
   context 'update' do
+    let(:existing_container) { FactoryGirl.create(container_sym) }
+    let(:job_transaction) { described_class.initialize_job(existing_container) }
     context 'perform_now' do
+      include_context 'tracking job', :job_transaction
       it {
-        existing_container = FactoryGirl.create(container_sym)
+        expect(existing_container).to be_persisted
         mocked_proxy = double()
         expect(mocked_proxy).to receive(:update_document)
         expect(existing_container).to receive(:__elasticsearch__).and_return(mocked_proxy)
-        ElasticsearchIndexJob.perform_now(existing_container, update: true)
+        ElasticsearchIndexJob.perform_now(job_transaction, existing_container, update: true)
       }
     end
   end
 
   context 'create' do
+    let(:new_container) { FactoryGirl.create(container_sym) }
+    let(:job_transaction) { described_class.initialize_job(new_container) }
     context 'perform_now' do
+      include_context 'tracking job', :job_transaction
       it {
-        new_container = FactoryGirl.create(container_sym)
+        expect(new_container).to be_persisted
         mocked_proxy = double()
         expect(mocked_proxy).to receive(:index_document)
         expect(new_container).to receive(:__elasticsearch__).and_return(mocked_proxy)
-        ElasticsearchIndexJob.perform_now(new_container)
+        ElasticsearchIndexJob.perform_now(job_transaction, new_container)
+        job_transaction.reload
```

Comment:
What is the call to `#reload` doing here? I'm confused because there aren't any expectations after it.